syntax = "proto3";

package createorlogin.v1;

enum ResponseStatus {
    SUCCESS = 0;                
    FAILURE = 1;                
    INVALID_INPUT = 2;          
    ACCOUNT_EXISTS = 3;         
    ACCOUNT_NOT_FOUND = 4;      
    UNAUTHORIZED = 5;           
}

message User {
    int32 user_id = 1;          
    string user_name = 2;      
    string display_name = 3;   
}

message CreateAccountRequest {
    string user_name = 1;      
    string display_name = 2;    
    string password = 3;       
}

message LoginAccountRequest {
    string user_name = 1;     
    string password = 2;       
}

message CreateAccountResponse {
    ResponseStatus status = 1;  
    User myself = 2;
    repeated User contacts = 3;         
}

message LoginAccountResponse {
    ResponseStatus status = 1;  
    User myself = 2;
    repeated User contacts = 3; 
       
}

service AccountService {
    rpc CreateAccount(CreateAccountRequest) returns (CreateAccountResponse);
    rpc LoginAccount(LoginAccountRequest) returns (LoginAccountResponse);
}